lib/msal-node/src/client/ClientApplication.ts

```diff
@@ -279,33 +282,132 @@ export abstract class ClientApplication {
             validRequest.correlationId,
             validRequest.forceRefresh
         );
+
         try {
-            const silentFlowClientConfig =
+            const clientConfiguration =
                 await this.buildOauthClientConfiguration(
                     validRequest.authority,
                     validRequest.correlationId,
                     validRequest.redirectUri || "",
                     serverTelemetryManager,
                     undefined,
-                    request.azureCloudOptions
+                    validRequest.azureCloudOptions
                 );
-            const silentFlowClient = new SilentFlowClient(
-                silentFlowClientConfig
-            );
+            const silentFlowClient = new SilentFlowClient(clientConfiguration);
             this.logger.verbose(
                 "Silent flow client created",
                 validRequest.correlationId
             );
-            return await silentFlowClient.acquireToken(validRequest);
+
+            try {
+                return await this.acquireCachedToken(
+                    validRequest,
+                    silentFlowClient,
+                    clientConfiguration
+                );
+            } catch (e) {
+                try {
+                    // If persistence exists, overwrite the in-memory cache with the persistent cache and search for the cached token again
+                    const persistence = this.config.cache.cachePlugin;
+                    if (persistence) {
```

Comment:
this.tokenCache is defined on line 108 with `this.config.cache.cachePlugin` as a parameter. I think you can use `this.tokenCache.persistence` instead of defining a new variable here.